tests/unit/amo/components/TestCollectionManager.js

```diff
@@ -567,8 +580,73 @@ describe(__filename, () => {
       root, propName: 'onSuggestionSelected',
     });
     selectSuggestion(suggestion);
-    // TODO: test onSuggestionSelected
-    // https://github.com/mozilla/addons-frontend/issues/4590
+
+    sinon.assert.calledWith(dispatchSpy, addAddonToCollection({
+      addonId: suggestion.addonId,
+      collectionId: collection.id,
+      collectionSlug: collection.slug,
+      editing: true,
+      errorHandlerId: errorHandler.id,
+      page,
+      userId: signedInUserId,
+    }));
+  });
+
+  it('dispatches addAddonToCollection with a default page of 1 when selecting an add-on', () => {
+    const errorHandler = createStubErrorHandler();
+
+    const collection = createInternalCollection({
+      detail: createFakeCollectionDetail({ authorUsername: signedInUsername }),
+    });
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const root = render({ collection, errorHandler, page: undefined });
+
+    const suggestion = createInternalSuggestion(
+      createFakeAutocompleteResult({ name: 'uBlock Origin' })
+    );
+    const selectSuggestion = simulateAutoSearchCallback({
+      root, propName: 'onSuggestionSelected',
+    });
+    selectSuggestion(suggestion);
+
+    sinon.assert.calledWith(dispatchSpy, addAddonToCollection({
+      addonId: suggestion.addonId,
+      collectionId: collection.id,
+      collectionSlug: collection.slug,
+      editing: true,
+      errorHandlerId: errorHandler.id,
+      page: 1,
+      userId: signedInUserId,
+    }));
+  });
+
+  it('sets the addAddonStatus state to pending when selecting an add-on', () => {
+    const root = render({});
+
+    const state = root.state();
+    expect(state.addAddonStatus).toEqual(null);
+
+    const suggestion = createInternalSuggestion(
+      createFakeAutocompleteResult({ name: 'uBlock Origin' })
+    );
+    const selectSuggestion = simulateAutoSearchCallback({
+      root, propName: 'onSuggestionSelected',
+    });
+    selectSuggestion(suggestion);
+
+    const newState = root.state();
+    expect(newState.addAddonStatus).toEqual(ADDON_ADDED_STATE_PENDING);
+  });
+
+  it('displays a notification after an add-on has been added', () => {
+    const root = render({});
+
+    expect(root.find(Notice)).toHaveLength(0);
+
+    root.setProps({ hasAddonBeenAdded: true });
```

Comment:
So what happens if `hasAddonBeenAdded` is `null` here?

Reply:
`hasAddonBeenAdded` is a boolean which does not accept nulls. I did add an extra test case to verify that the notification is removed when `hasAddonBeenAdded` switches from `true` to `false`.